File: SocketEdu-Python/models/turma.py
import json
from .aluno import Aluno 

class Turma:

    def __init__(self, id_turma, nome, ano):

        self.id = id_turma
        self.nome = nome
        self.ano = ano
        self.alunos = [] 

    def adicionar_aluno(self, aluno):
        self.alunos.append(aluno) 

    def listar_dados(self):

        print(f'\n{self.nome} ({self.ano})\n')

        for aluno in self.alunos:
            print(f'- {aluno.nome}, {aluno.idade} anos, matricula {aluno.matricula}')

    def to_json(self):

        turma_dict = {

            'id': self.id,
            'nome': self.nome,
            'ano': self.ano,
            'alunos': [
                {
                    'matricula': aluno.matricula,
                    'nome': aluno.nome,
                    'idade': aluno.idade
                }

                for aluno in self.alunos
            ]
        }

        return json.dumps(turma_dict)

File: SocketEdu-Java/src/models/Turma.java
package models;

import java.util.ArrayList;
import java.util.List;

public class Turma {

    private String id;
    private String nome;
    private int ano;
    private List<Aluno> alunos;

    public Turma(String id, String nome, int ano) {
        this.id = id;
        this.nome = nome;
        this.ano = ano;
        alunos = new ArrayList<>();
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public int getAno() {
        return ano;
    }

    public void setAno(int ano) {
        this.ano = ano;
    }

    public void adicionarAluno(Aluno aluno) {
        this.alunos.add(aluno);
    }

    public String toJSON() {

        StringBuilder json = new StringBuilder();
        json.append("{");
        json.append("\"id\":\"").append(id).append("\",");
        json.append("\"nome\":\"").append(nome).append("\",");
        json.append("\"ano\":").append(ano).append(",");
        json.append("\"alunos\":[");

        for (int i = 0; i < alunos.size(); i++) {
            json.append(alunos.get(i).toJSON());
            if (i < alunos.size() - 1) {
                json.append(",");
            }
        }

        json.append("]}");

        return json.toString();
    }
}
